gravitee-apim-cypress/cypress/integration/rest-api/apis/oauth2-policy-tests.ts

```diff
@@ -0,0 +1,352 @@
+          .then((application: Application) => {
+            am_oneScopeApplication = application;
+            const newTokenScopeSetting = {
+              settings: {
+                oauth: {
+                  enhanceScopesWithUserPermissions: false,
+                  scopeSettings: [
+                    {
+                      scope: 'scim',
+                      defaultScope: true,
+                    },
+                  ],
+                },
+              },
+            };
+            am_patchApplication(am_apiToken, am_domainId, am_oneScopeApplication.id, newTokenScopeSetting).ok();
+            const oauthConfig = { checkRequiredScopes: true, requiredScopes: ['scim'] };
+            createOauth2Api(am_domainHrid, application.settings.oauth.clientId, application.settings.oauth.clientSecret, oauthConfig)
+              .ok()
+              .its('body')
+              .then((createdApi) => {
+                oneScopeApi = createdApi;
+                publishPlan(API_PUBLISHER_USER, oneScopeApi.id, oneScopeApi.plans[0].id).ok();
+                deployApi(API_PUBLISHER_USER, oneScopeApi.id).ok();
+                startApi(API_PUBLISHER_USER, oneScopeApi.id).noContent();
+              });
+          });
+
+        am_createApplication(am_apiToken, am_expiredTokenApplicationName, am_domainId)
+          .its('body')
+          .then((application: Application) => {
+            am_expiredTokenApplication = application;
+            const newTokenValiditySetting = {
+              settings: {
+                oauth: {
+                  tokenCustomClaims: [],
+                  accessTokenValiditySeconds: 1,
+                  refreshTokenValiditySeconds: 14400,
+                  idTokenValiditySeconds: 14400,
+                },
+              },
+            };
+            am_patchApplication(am_apiToken, am_domainId, am_expiredTokenApplication.id, newTokenValiditySetting).ok();
+            createOauth2Api(am_domainHrid, application.settings.oauth.clientId, application.settings.oauth.clientSecret)
+              .ok()
+              .its('body')
+              .then((createdApi) => {
+                expiredTokenApi = createdApi;
+                publishPlan(API_PUBLISHER_USER, expiredTokenApi.id, expiredTokenApi.plans[0].id).ok();
+                deployApi(API_PUBLISHER_USER, expiredTokenApi.id).ok();
+                startApi(API_PUBLISHER_USER, expiredTokenApi.id).noContent();
+              });
+          });
+      });
+  });
+
+  describe('General access token tests', () => {
+    before(() => {
+      requestGateway(
+        {
+          method: 'POST',
+          url: `${Cypress.env('am_gatewayServer')}/auth/${am_domainHrid}/oauth/token`,
+          form: true,
+          auth: {
+            username: am_noScopeApplication.settings.oauth.clientId,
+            password: am_noScopeApplication.settings.oauth.clientSecret,
+          },
+          body: 'grant_type=client_credentials',
+        },
+        {
```

Comment:
Global comment for all occurrences of "validWhen" parameter ;

There is no need to override validWhen here, cause `response.status === 200` if the default validation rule in `requestGateway` function.

We could remove the second parameter.